### PR TITLE
Use cursors in `find` command

### LINKED ISSUE #1733: Support `getMore` command
### What should be done?

https://www.mongodb.com/docs/manual/reference/command/getMore/

- [x] Do not consume all documents in `find`: #2933
- [x] Do not consume all documents in `aggregate`
- [ ] Refactor tests (https://github.com/FerretDB/FerretDB/pull/2929/files#r1246365875)

Test script: #2761

### PR DESCRIPTION
**Pull request by AlekSi**
# Description

Closes #1733.

## Readiness checklist

- [x] I added/updated unit tests.
- [ ] I added/updated integration/compatibility tests.
- [x] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [x] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by w84thesun — APPROVED** [review]
LGTM

File: internal/handlers/pg/pgdb/transactions.go [hunk 1]
```diff
@@ -54,36 +54,58 @@ func (e *transactionConflictError) Error() string {
...
+	})
+	if err != nil {
+		err = lazyerrors.Error(err)
+		return
+	}
+
+	err = keepTx.Commit(ctx)
+	if err != nil {
+		err = lazyerrors.Error(err)
+		_ = keepTx.Rollback(ctx)
```
Comment (chilagrow) [inline internal/handlers/pg/pgdb/transactions.go:71]: On a rare occasion where rollback on commit fails, do we want to log the error here too?

File: internal/handlers/pg/msg_find.go [hunk 2]
```diff
@@ -121,22 +123,21 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
...
-		cancel()
+		closer.Close()
 		return nil, lazyerrors.Error(err)
 	}
 
-	iter := iterator.Values(iterator.ForSlice(resDocs))
-	closer := iterator.NewMultiCloser(iter, iterator.CloserFunc(cancel))
+	closer.Add(iterator.CloserFunc(func() {
+		// ctx could be cancelled already
+		_ = keepTx.Rollback(context.Background())
```
Comment (chilagrow) [inline internal/handlers/pg/msg_find.go:136]: What is the reason that we rollback in closer? Maybe because `find` should not modify data? :thinking:

**Review by chilagrow — COMMENTED** [review]
Just one question so I understand it :hugs:

File: internal/handlers/pg/msg_find.go (hunk 2 above)
Comment (AlekSi) [reply internal/handlers/pg/msg_find.go:136]: It does not matter if we commit transaction or rollback it, but we should close it

Comment (chilagrow) [reply internal/handlers/pg/msg_find.go:136]: Got it thanks :+1:

File: internal/handlers/pg/pgdb/transactions.go (hunk 1 above)
Comment (AlekSi) [reply internal/handlers/pg/pgdb/transactions.go:71]: Commit and rollback are already logged by our pgx logging configuration

**Comment by AlekSi** [comment]
@chilagrow PTAL

File: internal/handlers/pg/pgdb/transactions.go (hunk 1 above)
Comment (chilagrow) [reply internal/handlers/pg/pgdb/transactions.go:71]: I had no idea thanks :+1:

**Review by chilagrow — APPROVED** [review]
:rocket:

**Pull request merged**